### PR TITLE
net-misc/sslh-1.18-r1: new version that builds systemd generator and install systemd service

### PR DESCRIPTION
**Pull request by candrews**
https://bugs.gentoo.org/show_bug.cgi?id=563600

### CODE REVIEW

**Comment by mgorny** [comment]
Does the build system actually require systemd during build? Generators are usually quite plain C programs with no dep on systemd libraries.

**Comment by candrews** [comment]
@mgorny Yes, it does link against the systemd library, see https://github.com/yrutschle/sslh/blob/v1.18/Makefile

File: net-misc/sslh/sslh-1.18-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	systemd? ( sys-apps/systemd )
+	dev-libs/libconfig"
+DEPEND="${RDEPEND}
+	dev-lang/perl"
+
+RESTRICT="test"
+
+S=${WORKDIR}/${MY_P}
+
+PATCHES=( "${FILESDIR}/${PN}-1.17-version-deps.patch" "${FILESDIR}/${PN}-1.18-version-deps.patch" "${FILESDIR}/${PN}-1.18-systemd-generator.patch" )
```
Comment (SoapGentoo) [inline net-misc/sslh/sslh-1.18-r1.ebuild]: have one patch per line for readability

Comment (candrews) [inline net-misc/sslh/sslh-1.18-r1.ebuild]: @SoapGentoo thanks for the suggestion - I've made the change and pushed it.

**Comment by candrews** [comment]
@kensington is there anything else I can do or need to do in order to get this merged?

File: net-misc/sslh/sslh-1.18-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+SRC_URI="http://www.rutschle.net/tech/${PN}/${MY_P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~m68k ~mips ~s390 ~sh ~x86"
+IUSE="caps tcpd systemd"
+
+RDEPEND="caps? ( sys-libs/libcap )
+	tcpd? ( sys-apps/tcp-wrappers )
+	systemd? ( sys-apps/systemd )
```
Comment (kensington) [inline net-misc/sslh/sslh-1.18-r1.ebuild]: If sslh links to libsystemd.so, we could consider adding a subslot operator: ``systemd? ( sys-apps/systemd:= )``.

According to systemd's metadata.xml, the subslot will be incremented if there's an ABI break, so the operator will trigger a rebuild of sslh.

File: net-misc/sslh/sslh-1.18-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	dosym sslh-fork /usr/sbin/sslh
+	doman sslh.8.gz
+	dodoc ChangeLog README.md
+
+	newinitd "${FILESDIR}"/sslh.init.d-2 sslh
+	newconfd "${FILESDIR}"/sslh.conf.d-2 sslh
+
+	# Gentoo puts the binaries in /usr/sbin, but upstream puts them in /usr/bin
+	sed -i -e 's~/usr/bin/~/usr/sbin/~g' scripts/systemd.sslh.service || die
+	systemd_newunit scripts/systemd.sslh.service sslh.service
```
Comment (kensington) [inline net-misc/sslh/sslh-1.18-r1.ebuild]: Is the unit useful without the generator? If not, it's probably better to make the above two lines conditional too.

File: net-misc/sslh/sslh-1.18-r1.ebuild [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+	dev-libs/libconfig"
+DEPEND="${RDEPEND}
+	dev-lang/perl"
+
+RESTRICT="test"
+
+S=${WORKDIR}/${MY_P}
+
+PATCHES=(
+	"${FILESDIR}/${PN}-1.17-version-deps.patch"
```
Comment (kensington) [inline net-misc/sslh/sslh-1.18-r1.ebuild]: I've had trouble with this issue for a while (I can only ever reproduce it the first time, I guess due to caching), but your ``sslh-1.18-version-deps.patch`` seems to be a better fix so I would consider dropping the 1.17 version.

**Comment by kensington** [comment]
@candrews sorry for taking so long to looks at this, and thanks a lot for your work.

This looks pretty good. I've added a couple of inline comments and would consider revising the commit message to something like:

```
net-misc/sslh: revision bump adding systemd support

This new revision builds the systemd generator and installs the systemd service.

It also renames OPTIONS to DAEMON_OPTS in the sslh conf.d file for compatibility with the systemd service which sources it.

Gentoo-bug: 563600
```

**Comment by candrews** [comment]
@kensington I've incorporated your feedback, except for the patch. I also have sometimes gotten errors, but I'm not familiar enough with the build system to find / really fix the issue. Do you have thoughts on what else we can do? Or are we good as it is, and can handle such fixes in a separate bug / PR?

**Comment by kensington** [comment]
@candrews Thanks, picked as f51fc3240bc6f1f37736ece7f8ecfc5a242e7712. I also squashed in an alphabetical sorting of inherit, IUSE and RDEPEND which I didn't notice before.

As for the build issue, we can always look into it again if upstream ever responds.

**Pull request closed without merging**